### PR TITLE
BUG: Fix preset state readbacks

### LINKED ISSUE #955: Preset States Readback is Wrong
## Expected Behavior

expecting motor.presets.state() -> text name of the current state

## Current Behavior

motor.presets.state() -> not necessarily the state name

## Context

Reported by Diling at XPP via observing the non-changing status prints

### PR DESCRIPTION
**Pull request by ZLLentz**
## Description

Fix the issue where preset state readbacks were wrong.

## Motivation and Context

These state readbacks show up in the terminal pretty printer and are confusing if incorrect.
closes #955 

## How Has This Been Tested?

Added a test to demonstrate the failure, fixed the code appropriately, test now passes.

## Where Has This Been Documented?

Only in release notes

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [x] Test suite passes on travis
- [x] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Review by ZryletTC — COMMENTED** [review]
Looks good but I'm not sure the assertions actually test the problem case.

**Comment by ZLLentz** [comment]
The assertions failed before the change

File: pcdsdevices/interface.py [hunk 1]
```diff
@@ -1234,7 +1234,7 @@ def state(self):
             if method_name.startswith('wm_'):
                 state_name = method_name.replace('wm_', '', 1)
                 wm_state = getattr(device, method_name)
-                diff = wm_state()
+                diff = abs(wm_state())
                 if diff < closest:
```
Comment (tangkong) [inline pcdsdevices/interface.py:1238]: closest = 0.5 seems like a large margin for error, but it's not really a show stopper

**Review by tangkong — APPROVED** [review]
Looks good to me, neat functionality

**Review by ZryletTC — APPROVED** [review]
Nevermind, I was being dumb and didn't see the full extent of the issue. This makes sense after looking at it again 👍

**Pull request merged**